Expose charsets as Hash in normalized params

heidi opened the pull request:
Based on https://github.com/ygnessin/griddler/pull/4, in tandem with https://github.com/apartmentlist/griddler/pull/1

Parse the JSON string and return a Hash with symbol keys
for ease of use by Griddler later on.

heidi committed:
Expose charsets as Hash in normalized params

Parse the JSON string and return a Hash with symbol keys
for ease of use by Griddler later on.

tcollier commented on spec/griddler/sendgrid/adapter_spec.rb:
```diff
@@ -129,13 +137,38 @@
+
+    normalize_params(params)[:charsets].should be_nil
+  end
   def default_params
```
👾

tcollier commented on spec/griddler/sendgrid/adapter_spec.rb:
```diff
@@ -14,6 +14,14 @@
+        subject: 'UTF-8',
+        from: 'UTF-8',
+        html: 'UTF-8',
+        text: 'iso-8859-1'
```
You don't gain much in this test by having an exhaustive list of `charsets`. I suggest only having 2, with different values. Then you can do away with the loop in the spec.

heidi replied:
👍

tcollier commented on spec/griddler/sendgrid/adapter_spec.rb:
```diff
@@ -129,13 +137,38 @@
       cc: 'cc@example.com',
       from: 'there@example.com',
       envelope: "{\"to\":[\"johny@example.com\"], \"from\": [\"there@example.com\"]}",
+      charsets: {
```
No specs are using this, so I suggest leaving it out.

heidi replied:
The `it 'returns the charsets as a hash'` spec is using it.

tcollier commented on lib/griddler/sendgrid/adapter.rb:
```diff
@@ -40,6 +41,15 @@ def bcc_from_envelope
+          JSON.parse(params[:charsets]).symbolize_keys
+        end
+      rescue JSON::ParserError
+        nil
```
Did you consider defaulting to a `{}`?

heidi replied:
Not originally, but I agree: `{}` is better than `nil`.

heidi committed:
Don't list full set of charsets

It's not adding anything to our specs. We can see the full
list of fields here if needed:
https://sendgrid.com/docs/API_Reference/Webhooks/inbound_email.html#-Character-Sets-and-Header-Decoding

heidi committed:
Default to {} whe charsets unspecified

This lets us deal with an empty hash instead of having to
check for nil when working with charsets
